modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts

```diff
@@ -75,35 +112,64 @@ export async function annualRule(
 	currentDateString: string[],
 	january: string,
 	yearToDateGoalsArray: string[],
-	yearToDateSalesArray: string[]
+	yearToDateSalesArray: string[],
+	productExternalReferenceCode: string
 ) {
 	const salesGoal = await getSalesGoal(
 		currentDateString[0],
 		currentDateString[1],
 		currentDateString[0],
 		january
 	);
-	const yearToDateGoalsResult = getArrayOfGoals(
-		salesGoal,
-		yearToDateGoalsArray
-	);
 
 	const policiesSalesGoals = await getPoliciesForSalesGoal(
 		currentDateString[0],
 		currentDateString[1],
 		currentDateString[0],
 		january
 	);
-	const yearToDateSalesResult = getArrayOfSales(
-		policiesSalesGoals,
-		yearToDateSalesArray
+
+	const productsType = salesGoal.data.items.filter((item: SalesGoal) => {
```

Comment:
How are you using this more than once, maybe you should put in a variable like
```suggestion
	const getSalesGoalData = salesGoal?.data?.items
```